src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -722,5 +866,17 @@ private int GetNumericEnd(string path)
 
             return -1;
         }
+
+        private CheckNumericEndResult GetNumericEndResult(string path)
+        {
+            var possibleIndex = path.Substring(path.LastIndexOf("/") + 1);
```

Comment:
Yeah, this is what I expected when I looked at the other code... You'll later use the same logic to split a path like `/foo/bar/5` into `/foo/bar` and `5`. You may as well do it here and make it part of the result.